src/core_plugins/kibana/public/discover/embeddable/search_embeddable_handler.js

```diff
@@ -0,0 +1,77 @@
+import searchTemplate from './search_template.html';
+import angular from 'angular';
+import * as columnActions from 'ui/doc_table/actions/columns';
+import { getPersistedStateId } from 'plugins/kibana/dashboard/panel/panel_state';
+import { EmbeddableHandler } from 'ui/embeddable';
+
+
+export class SearchEmbeddableHandler extends EmbeddableHandler {
+
+  constructor($compile, $rootScope, searchLoader) {
+    super();
+    this.$compile = $compile;
+    this.searchLoader = searchLoader;
+    this.$rootScope = $rootScope;
+    this.name = 'search';
+  }
+
+  async getEditPath(panelId) {
+    return await this.searchLoader.urlFor(panelId);
```

Comment:
If an embeddable had controls for editing itself it would need to be able to tell consumers that the title/path/etc. have changed, any idea how that would work?

I assume we couldn't just replace these methods with something like `ContainerAPI#setEditPath()` and `ContainerAPI#setTitle()` because we want to show the edit path and title for a panel without rendering it... Maybe the container API could expose a method to invalidate the editPath and titles that are cached by containers?